packages/react-liveness/src/components/FaceLivenessDetector/service/utils/__tests__/getFaceMatchStateInLivenessOval.test.ts

```diff
@@ -0,0 +1,112 @@
+import 'jest-canvas-mock';
+import { getFaceMatchStateInLivenessOval } from '../getFaceMatchStateInLivenessOval';
+import {
+  mockCloselyMatchedFace,
+  mockMatchedFace,
+  mockOffCenterFace,
+  mockTooFarFace,
+  mockOvalDetails,
+  mockSessionInformation,
+} from '../__mocks__/testUtils';
+import { Face, FaceMatchState, LivenessOvalDetails } from '../../types';
+import { SessionInformation } from '@aws-sdk/client-rekognitionstreaming';
+
+describe('getFaceMatchStateInLivenessOval', () => {
+  it('should throw an error if the challenge config is undefined', () => {
+    const face: Face = mockOffCenterFace;
+    const ovalDetails: LivenessOvalDetails = mockOvalDetails;
+    const initialFaceIntersection: number = 0.3;
+    const undefinedConfigSessionInformation: SessionInformation = {
+      ...mockSessionInformation,
+      // @ts-expect-error
+      Challenge: { FaceMovementAndLightChallenge: {} },
+    };
+
+    expect(() => {
+      getFaceMatchStateInLivenessOval({
+        face,
+        ovalDetails,
+        initialFaceIntersection,
+        sessionInformation: undefinedConfigSessionInformation,
+        frameHeight: 480,
+      });
+    }).toThrowError(
+      'Challenge information not returned from session information.'
+    );
+  });
+
+  it('should parse sessionInformation and return oval parameter attributes', () => {
+    const face: Face = mockOffCenterFace;
+    const ovalDetails: LivenessOvalDetails = mockOvalDetails;
+    const initialFaceIntersection: number = 0.3;
+    const sessionInformation = mockSessionInformation;
+
+    const { faceMatchState, faceMatchPercentage } =
+      getFaceMatchStateInLivenessOval({
+        face,
+        ovalDetails,
+        initialFaceIntersection,
+        sessionInformation,
+        frameHeight: 480,
+      });
+
+    expect(faceMatchState).toBe(FaceMatchState.OFF_CENTER);
+    expect(faceMatchPercentage).toBe(0);
+  });
+
+  it('should return MATCHED when face is close and in frame', () => {
+    const face: Face = mockCloselyMatchedFace;
+    const ovalDetails: LivenessOvalDetails = mockOvalDetails;
+    const initialFaceIntersection: number = 0.3;
+    const sessionInformation = mockSessionInformation;
+
+    const { faceMatchState } = getFaceMatchStateInLivenessOval({
+      face,
+      ovalDetails,
+      initialFaceIntersection,
+      sessionInformation,
+      frameHeight: 480,
+    });
+
+    expect(faceMatchState).toBe(FaceMatchState.MATCHED);
+  });
+
+  // Note: if this test ever fails due to face detection updates just add a console log for face and do a check and copy the value
```

Comment:
What can we do to improve the tests/mocks to avoid needing this comment?

Reply:
It would be nice if we could generate a mocked face based on the constraints, like reverse engineer the matching algorithm to create a face that falls within the criteria for MATCH/TOO_FAR etc. But that would need to be updated whenever you update the original algorithm so it might be more work 🤔

Reply:
> But that would need to be updated whenever you update the original algorithm so it might be more work

Think that's expected when you make changes though? Like tests **should** break when we change values